src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -330,6 +330,12 @@ protected void validateSchema(Table table, Schema tableSchema, boolean allowSche
       return;
     }
 
+    if (isTruncateTableSet()) {
+      //no validation required for schema if truncate table is set.
```

Comment:
why isn't this needed? BQ will change the schema of the table change after a truncate? If so, add that information to the comment.

Reply:
Yes, truncate table sets writeDisposition as WRITE_TRUNCATE and this overwrites the schema. I will add details to comment

Reply:
Done